### ISSUE TITLE
[Dialog] Add `label` property

### ISSUE THREAD

**Issue by ellenupp** [Esri/calcite-design-system#12300]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

The dialog component doesn't have an explicit label property but setting `heading` will set the aria-label as well. This makes sense when you want that visible heading, but when creating a custom header using the `header-content` slot it might be clearer to have a specific `label`. This could also be used to set a label different from the heading. 

### Acceptance Criteria

Dialog has a way to specifically set its aria-label. 

### Which Component

dialog

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Dashboards
